tool/site/client/src/components/usePagination.tsx

```diff
@@ -0,0 +1,77 @@
+import {useMemo} from "react"
+
+export const DOTS = "..."
+
+const range = (start, end) => {
+  const length = end - start + 1
+  return Array.from({length}, (_, idx) => idx + start)
+}
+
+interface UsePaginiationProps {
+  totalCount: number
+  pageSize: number
+  siblingCount: number
+  currentPage: number
+}
+
+export const usePagination = ({
+  totalCount,
+  pageSize,
+  siblingCount,
+  currentPage
+}: UsePaginiationProps): Array<string | number> => {
```

Comment:
I think we can use `Array<typeof DOTS | number>` then get rid of the casting we do in `pagination.tsx`

Reply:
Yep that's much better. Thanks